miden-tx/src/errors/mod.rs

```diff
@@ -44,18 +44,19 @@ impl std::error::Error for TransactionExecutorError {}
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum TransactionProverError {
-    ProveTransactionProgramFailed(ExecutionError),
+    InternalError(String),
     InvalidAccountDelta(AccountError),
     InvalidTransactionOutput(TransactionOutputError),
     ProvenTransactionError(ProvenTransactionError),
+    TransactionExecutionProgramFailed(ExecutionError),
```

Comment:
minor nit (feel free to ignore): to me, `TransactionProgramExecutionFailed` sounds a bit more correct as it is program execution that fails.

Reply:
Thanks, that is what I meant to write.